usual/crypto/md5_openssl.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Using MD5 OPENSSL APIs.
+ */
+#include <usual/crypto/md5.h>
+#include <usual/crypto/digest.h>
+
+#ifdef OPENSSL_API_COMPAT
+static void md5_crypt_cleanup(struct md5_ctx *ctx)
+{
+	ctx->errreason = ERR_reason_error_string(ERR_get_error());
+	ctx->error = MD5_ERROR_OPENSSL;
+	EVP_MD_CTX_destroy(ctx->evpctx);
+	ctx->evpctx = NULL;
+	ERR_clear_error();
+}
+
+static void md5_init(struct md5_ctx *ctx)
+{
+	if (ctx->evpctx != NULL)
+	{
+		ctx->error = MD5_ERROR_NONE;
+		ctx->errreason = NULL;
+		if (EVP_DigestInit_ex(ctx->evpctx, EVP_md5(), NULL) <= 0)
+			md5_crypt_cleanup(ctx);
+	}
+}
+void md5_reset(struct md5_ctx *ctx)
+{
+	ERR_clear_error();
+	ctx->evpctx = EVP_MD_CTX_create();
```

Comment:
Should check if `evpctx` is NULL, if not we should call `md5_crypt_cleanup` on it before assigning a new value to it.